financial-connections/src/main/java/com/stripe/android/financialconnections/core/FinancialConnectionsViewModel.kt

```diff
@@ -23,31 +24,28 @@ internal abstract class FinancialConnectionsViewModel<S>(
     val stateFlow: StateFlow<S> = _stateFlow.asStateFlow()
 
     protected open fun <T : Any?> (suspend () -> T).execute(
-        onSuccess: (T) -> Unit = {},
-        onFail: (Throwable) -> Unit = {},
-        reducer: S.(Async<T>) -> S
+        retainValue: KProperty1<S, Async<T>>? = null,
+        reducer: S.(Async<T>) -> S,
     ): Job {
         return viewModelScope.launch {
-            setState { reducer(Loading) }
+            setState { reducer(Loading(value = retainValue?.get(this)?.invoke())) }
```

Comment:
Took this from mavericks: This allows us to optionally keep the current Async value if moving to a Loading state from Success (so that we can keep the screen content while loading)